mlir/lib/Dialect/QCO/Transforms/Optimizations/QuantumFunctionBoundaryCommutation.cpp

```
@@ -0,0 +1,116 @@
+  auto newUser = dyn_cast<UnitaryOpInterface>(*newParameter.getUsers().begin());
+
+  for (auto i = 0U; i < newUser.getNumQubits(); ++i) {
+    newUser.getOutputQubit(i).replaceAllUsesWith(newUser.getInputQubit(i));
+  }
+  newUser.erase();
+  previousSpecializations[funcOp.getName().str()] = newFunc;
+
+  call.setCallee(newFunc.getName());
+}
+
+void runQuantumFunctionBoundaryCommutation(ModuleOp module,
```

Cppcheck (2.21.0)

[style] The function 'runQuantumFunctionBoundaryCommutation' is never used.

(unusedFunction)